stl/inc/span

```diff
@@ -0,0 +1,665 @@
+
+    constexpr explicit _Span_extent_type(const size_t _Size) noexcept : _Mysize(_Size) {}
+
+    template <size_t _OtherExtent>
+    constexpr explicit _Span_extent_type(const _Span_extent_type<_OtherExtent> _Ext) noexcept : _Mysize(_Ext._Size()) {}
+
+    _NODISCARD constexpr size_t size() const noexcept {
+        return _Mysize;
+    }
+
+private:
+    size_t _Mysize{0};
+};
+
+// VARIABLE TEMPLATE _Is_span_v
+template <class _Ty, size_t _Extent>
+class span;
+
+template <class>
+inline constexpr bool _Is_span_v = false;
+
+template <class _Ty, size_t _Extent>
+inline constexpr bool _Is_span_v<span<_Ty, _Extent>> = true;
+
+// VARIABLE TEMPLATE _Is_std_array_v
+template <class>
+inline constexpr bool _Is_std_array_v = false;
+
+template <class _Ty, size_t _Extent>
+inline constexpr bool _Is_std_array_v<array<_Ty, _Extent>> = true;
+
+// VARIABLE TEMPLATE _Is_span_convertible_range_v
+template <class _Cont, class _Ty>
+inline constexpr bool _Is_span_convertible_range_v =
+    is_convertible_v<remove_pointer_t<decltype(_STD data(_STD declval<_Cont&>()))> (*)[], _Ty (*)[]>;
+
+// VARIABLE TEMPLATE _Has_container_interface_v
+template <class, class = void>
+inline constexpr bool _Has_container_interface_v = false;
+
+template <class _Cont>
+inline constexpr bool _Has_container_interface_v<_Cont,
+    void_t<decltype(_STD data(_STD declval<_Cont&>())), decltype(_STD size(_STD declval<_Cont&>()))>> = true;
+
+// VARIABLE TEMPLATE _Is_span_compatible_container_v
+// clang-format off
+template <class _Cont, class _Ty>
+inline constexpr bool _Is_span_compatible_container_v =
+       !_Is_span_v<remove_const_t<_Cont>>
+    && !_Is_std_array_v<remove_const_t<_Cont>>
+    && !is_array_v<_Cont>
+    && _Is_span_convertible_range_v<_Cont, _Ty>
+    && _Has_container_interface_v<_Cont>;
+// clang-format on
+
+// [views.span]
+// CLASS TEMPLATE span
+template <class _Ty, size_t _Extent = dynamic_extent>
+class span : public _Span_extent_type<_Extent> {
+public:
+    using _Mybase                = _Span_extent_type<_Extent>;
+    using element_type           = _Ty;
+    using value_type             = remove_cv_t<_Ty>;
+    using index_type             = size_t;
+    using difference_type        = ptrdiff_t;
+    using pointer                = _Ty*;
+    using const_pointer          = const _Ty*;
+    using reference              = _Ty&;
+    using const_reference        = const _Ty&;
+    using iterator               = _Span_iterator<_Ty>;
+    using const_iterator         = _Span_iterator<const _Ty>;
+    using reverse_iterator       = _STD reverse_iterator<iterator>;
+    using const_reverse_iterator = _STD reverse_iterator<const_iterator>;
+
+    static constexpr index_type extent = _Extent;
+
+    // [span.cons] Constructors, copy, and assignment
+    constexpr span() noexcept = default;
+
+    constexpr span(pointer _Ptr, index_type _Count) noexcept // strengthened
+        : _Mybase(_Count), _Mydata(_Ptr) {}

```

Comment:
Missing validation for https://eel.is/c++draft/span.cons#3 . (The "valid range" part can't be checked, but the `_Count` precondition definitely can.)

Reply:
True, I had that check in `_Span_storage` if i remember correctly but forgot to adapt